src/api/sockets.c

```diff
@@ -1195,6 +1210,40 @@ lwip_recvfrom_udp_raw(struct lwip_sock *sock, int flags, struct msghdr *msg, u16
   return ERR_OK;
 }
 
+#if LWIP_NETPACKET
+static ssize_t
+lwip_recvfrom_netpacket(struct lwip_sock *sock, void *mem, size_t len, int flags, struct sockaddr *from, socklen_t *fromlen)
+{
+  struct netbuf *buf = NULL;
+  int err = netconn_recv(sock->conn, (struct netbuf **)&buf);
```

Comment:
not sure if we should/could add nonblock implementation here (MSG_DONTWAIT)

Reply:
This implementation of packet sockets is specifically tailored to our wpa_supplicant use case which currently doesn't require MSG_DONTWAIT. Original netpacket code in old kernel also doesn't support it. I've added TODO comment -  it will be required for Wi-Fi.

Reply:
ack, merging.